### PR TITLE
feat: [CI-6608]: HostNames for k8s infra

### PR DESCRIPTION
**Pull request by shaurya-harness**
### Summary

Added HostNames field for k8s infra in CI pipelines

#### Screenshots

Screenshot 2023-01-05 at 3 38 43 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx [hunk 1]
```diff
@@ -714,6 +718,13 @@ export default function BuildInfraSpecifications({ children }: React.PropsWithCh
           const filteredLabels = getMapValues(
             Array.isArray(values.labels) ? values.labels.filter((val: any) => testLabelKey(val.key)) : values.labels
           )
+          let filteredHostNames
+          if (values.hostNames && values.hostNames.length > 0) {
+            filteredHostNames =
+              typeof values.hostNames === 'string'
+                ? values.hostNames
+                : values.hostNames.map((listValue: { id: string; value: string }) => listValue.value)
+          }
```
Comment (d1wilko) [inline src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx:727]: `filteredHostNames` will sometimes be a string and sometimes be an array - is that ok?

no issues, just want to confirm this was the intention :)

**Review by d1wilko — DISMISSED** [review]
one open question - but approving PR assuming it is intentional :)

File: src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx (hunk 1 above)
Comment (shaurya-harness) [reply src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx:727]: that is for the case of runtime Inputs, we set it as <+input> string

**Review by kapilkumawat86 — DISMISSED** [review]

**Review by pankajsingh-harness — DISMISSED** [review]

**Review by vardanbansal-harness — DISMISSED** [review]

**Comment by shaurya-harness** [comment]
retigger jest

**Comment by shaurya-harness** [comment]
retrigger jest

**Review by d1wilko — APPROVED** [review]
LGTM

**Pull request merged**